test/libyul/YulInterpreterTest.cpp

```diff
@@ -130,12 +130,20 @@ string YulInterpreterTest::interpret()
 {
 	InterpreterState state;
 	state.maxTraceSize = 10000;
+	state.maxSteps = 10000;
+	state.maxMemSize = 0x20000000;
 	Interpreter interpreter(state);
 	try
 	{
 		interpreter(*m_ast);
 	}
-	catch (InterpreterTerminated const&)
+	catch (ExplicitlyTerminated const&)
```

Comment:
If `InterpreterTerminated` would be a common base class of the others, you would only need a single catch statement.

Reply:
so you are suggestion something like

```
try {
  ...
}
catch (InterpreterTerminated const& it)
{
  if (dynamic_cast<StepLimitReached>(it))
    return 0;
}
```

Reply:
Exactly! Your example above could also be:
```
try
{
...
}
catch (StepLimitReached const& _e)
{
...
}
catch (InterpreterTerminated const& _e)
{
... // all other cases
}
```
Perhaps the type should be named something like `InterpreterTerminatedGeneric` to make it more obvious that it is a generic base class.

Reply:
done.